src/idl_gen_general.cpp

```diff
@@ -378,7 +382,8 @@ class GeneralGenerator : public BaseGenerator {
 
   // Casts necessary to correctly read serialized data
   std::string DestinationCast(const Type &type) const {
-    if (type.base_type == BASE_TYPE_VECTOR) {
+    if (type.base_type == BASE_TYPE_VECTOR ||
```

Comment:
Optional:
`IsArray(type) || IsVector(type)` - ?
Or-short: `IsSeries(type)`

Here and whenever direct comparisons with `BASE_TYPE_VECTOR` and/or `BASE_TYPE_ARRAY` are used.

Reply:
Introduced `IsSeries` and `IsVector`, though I did not change existing comparisons against `BASE_TYPE_VECTOR`.